project/data/sample_threads.json

```diff
@@ -1,201 +1,162 @@
-[{
-        "model": "api.thread",
+[
+    {
+        "model": "threads.Thread",
         "fields": {
             "title": "Agriculture Subsidization ",
             "summary": "The United States currently pays around $20 billion per year to farmers in direct subsidies as \"farm income stabilization\" via farm bills. These bills pre-date the economic turmoil of the Great Depression with the 1922 Grain Futures Act, the 1929 Agricultural Marketing Act, and the 1933 Agricultural Adjustment Act creating a tradition of government support. The beneficiaries of the subsidies have changed as agriculture in the United States has changed. In the 1930s, about 25% of the country's population resided on the nation's 6,000,000 small farms. By 1997, 157,000 large farms accounted for 72% of farm sales, with only 2% of the US population residing on small farms.   In 2006, the top 3 states receiving subsidies were Texas (10.4%), Iowa (9.0%), and Illinois (7.6%). United States Department of Agriculture (USDA) subsidies from farms in Iowa totaled $1,212,000,000 in 2006. From 2003 to 2005, the top 1% of beneficiaries received 17% of subsidy payments. In Texas, 72% of farms do not receive government subsidies. Of the close to USD$1.4 billion in subsidy payments to farms in Texas, roughly 18% of the farms receive a portion of the payments.Top states for direct payments were Iowa ($501 million), Illinois ($454 million), and Texas ($397 million). Direct payments of subsidies are limited to $40,000 per person or $80,000 per couple.  The subsidy programs give farmers extra money for their crops and guarantee a price floor. For instance, in the 2002 Farm Bill, for every bushel of wheat sold, farmers were paid an extra 52 cents and guaranteed a price of 3.86 from 2002-03 and 3.92 from 2004-2007. That is, if the price of wheat in 2002 was $3.80, farmers would get an extra 58å¢ per bushel (52å¢ plus the 6å¢ price difference).  Acknowledging  https://en.wikipedia.org/wiki/Agricultural_subsidy. ",
-            "category_id": "1",
```

Comment:
Could you restore `category_id` and `author_id` fields? Both category and author fields have `null=True` in their relations but it is more appropriate for a `thread` to have a `category` and an `author`.

Reply:
@AbhijithGanesh, sorry for my late response. I was cloning your repo and look at your project. A thread should have these fields.

Reply:
Could we continue the conversation here?

What do you mean by `removed`?

Thread model has these fields.

```python
class Thread(models.Model):
    author = models.ForeignKey(
        Profile, default=None, null=True, on_delete=models.PROTECT
    )
    category = models.ForeignKey(
        Category, default=None, null=True, on_delete=models.PROTECT
    )
```

Let's change our perspective. Suppose one decides to create a new thread. What will this person do when creating a new thread? This person will write a title, a summary and choose a related category. In the end, the new thread will have all these fields. What we do with `fixtures` has to be the same. We just automize instance creation steps.

Reply:
I agree with the part which relates to "It has to be the same ". But threads and categories donot have a categories_id right? How can I implement it if they don't exist? Should I query it first using dbshell and figure a way around or add a new column?

Reply:
@gorkemarslan can you please help me with  this? I am a bit unsure about what to do

Reply:
@AbhijithGanesh, I'm typing. I don't just follow your speed. 😅  Sorry my bad

Reply:
😂😂😂 I am sorry but I was way too unsure about  to what to do and I really love this project

Reply:
Whenever you create a new model instance. The instance has `id` (or `pk` field), which is an automatic increase integer number. Whenever you create a model instance on the database,  a new id is automatically assigned. The `id` field is a hidden field but can be overridden. As far as databases are concerned, id fields are a must. Think of your ID number. You're a person with an ID number for your government (in this case the database).  We call you with your name but you're registered in the system with your id. The ID number is meaningless for your friends (in this case the users of the website). On the other hand, officers (in this case backend developers) are interested in your ID number. 

You can look the [docs](https://docs.djangoproject.com/en/3.2/topics/db/models/#automatic-primary-key-fields), as well.

Reply:
```python
class Category(models.Model):
    name = models.CharField(max_length=63)

    created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
    last_modified = models.DateTimeField(auto_now=True, blank=True, null=True)
```

So a category model instance has four fields. `id`, `name`, `created`, and `last_modified`.

Reply:
For example,
in **categories.json** file (in my branch it is not updated):
```json
{
    "fields": {
        "name": "Environment"
    },
    "model": "api.category",
    "pk": 1
}
```
**sample_threads.json:**
```json
{
        "model": "api.thread",
        "fields": {
            "title": "Agriculture Subsidization ",
            "summary": "...",
            "category_id": "1",
            "author_id": "2",
            "created": "2016-10-09 20:25:20.365392+00"
        }
    }
```
So this thread has a category with an id 1. (`"category_id": "1"`). You can see `categories.json` file, `categories_id` means `Environment` category.

Reply:
I understood where I went wrong, instead of removing author id I removed both the ids

Reply:
@gorkemarslan I have patched my mistake, thanks for the amazing catch😄😅

Reply:
Can you approve this @gorkemarslan

Reply:
@AbhijithGanesh, `author_id` field should also be included. Could you restore it?

Reply:
How will we map the author id to author? We don't know who wrote them right?

Reply:
Is there a way to map the authors to their ids Gorkem Arslan?

Reply:
@AbhijithGanesh, authors have already mapped Profile model via `author_id` field. 

```python
author = models.ForeignKey(
        Profile, default=None, null=True, on_delete=models.PROTECT
    )
```

So you should make any changes about `id` fields in the fixture data.

Reply:
> Is there a way to map the authors to their ids Gorkem Arslan? @gorkemarslan 
> 
> 
They are mapped but they don't actually exist , if they don't exist, we can't load them